log_system_information.py

```diff
@@ -0,0 +1,49 @@
+import json
+import subprocess
+import sys
+from pip._internal.operations.freeze import freeze
+import platform
+
+try:
+    import usb.core
+except ImportError:
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "pyusb"])
+    import usb.core
+
+
+def get_usb():
+    speeds = ["Unknown", "Low", "Full", "High", "Super"]
+    format_hex = lambda val: f"{val:#0{6}x}"
+    try:
+        for dev in usb.core.find(find_all=True):
+            yield {
+                "port": dev.port_number,
+                "vendor_id": format_hex(dev.idVendor),
+                "product_id": format_hex(dev.idProduct),
+                "speed": speeds[dev.speed]
```

Comment:
Need to extend the `speeds` array. Newer libusb versions also report `LIBUSB_SUPER_SPEED_PLUS`. I have installed the latest on my system and it crashes with:
`IndexError: list index out of range`

Reply:
Also to be future proof should be something like :
```
"speed": speeds[dev.speed] if dev_speed < len(speeds) else dev_speed
```